pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/SourceEditorController.java

```diff
@@ -126,6 +139,63 @@ public void refreshAST() {
     }
 
 
+    public void showAuxClassPathController(DesignerRoot root) {
+        AuxClassPathController auxClassPathController = new AuxClassPathController(root);
+
+        FXMLLoader fxmlLoader = new FXMLLoader(DesignerUtil.getFxml("auxclasspath-setup-popup.fxml"));
+
+        fxmlLoader.setControllerFactory(type -> {
+            if (type == AuxClassPathController.class) {
+                return auxClassPathController;
+            } else {
+                throw new IllegalStateException("Wrong controller!");
+            }
+        });
+        try {
+            Parent root1 = fxmlLoader.load();
+
+            auxClassPathController.setAuxclasspathFiles(auxclasspathFiles.getValue());
+
+            Stage stage = new Stage();
+            stage.initOwner(root.getMainStage());
+            stage.initModality(Modality.WINDOW_MODAL);
+            stage.setScene(new Scene(root1));
+            stage.show();
+
+            auxClassPathController.setOnApply(files -> {
+                stage.close();
+                auxclasspathFiles.setValue(files);
+            });
+
+            auxClassPathController.setOnCancel(stage::close);
+
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+    }
+
+
+    @PersistentProperty
+    public String getAuxclasspathFiles() {
+
+        StringBuilder sb = new StringBuilder();
+
+        for (File f : auxclasspathFiles.getValue()) {
+            sb.append(';').append(f.getAbsolutePath());
+        }
+        return sb.length() > 0 ? sb.substring(1) : "";
+    }
+
+
+    public void setAuxClassPathFiles(String files) {
+        List<File> newVal = Arrays.asList(files.split(";")).stream().map(File::new).collect(Collectors.toList());
+        auxclasspathFiles.setValue(newVal);
+    }
```

Comment:
you should here initialize a `ClasspathClassLoader` for `auxclasspathClassLoader` and have the `ASTManager` refresh everything using that classloader.